src/main/java/sdmExam/Board.java

```diff
@@ -76,7 +76,88 @@ public boolean existsDiagonallyAdjacentWithStone(Intersection intersection, Ston
                 .anyMatch(diagonalIntersection -> diagonalIntersection.hasStone(stone));
     }
 
+    public ArrayList<ArrayList<Intersection>> findTerritories() {
+        ArrayList<ArrayList<Intersection>> territories = new ArrayList<>();
+        ArrayList<Intersection> territory = new ArrayList<>();
+
+        for(int col = 1; col <= Board.BOARD_SIZE; col++){
+            Intersection intersection = new Intersection(Position.in(3, col), Stone.NONE);
+            territory.add(intersection);
+        }
+
+        territories.add(territory);
+        return territories;
+    }
+
+    public ArrayList<ArrayList<Intersection>> findRegions() {
+        ArrayList<ArrayList<Intersection>> regions = new ArrayList<>();
+
+        int h = intersections.size();
+        if (h == 0)
+            return regions;
+
+        for (int i = 0; i < h; i++) {
+            if (!intersections.get(i).isOccupied() && findIntersectionInRegions(regions, intersections.get(i))) {
```

Comment:
intersections.get(i).isOccupied() is tending to code smell, we may need to find another way to perform